File: Npc/11004302.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11004302: Ghost
/// </summary>
public class _11004302 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:1002141907011418$
    // - I'mma scratch <i>everything!</i>
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:1002141907011422$
                // - Now I'mma ghost, I can talk to people! How fun!
                return 30;
            case (30, 1):
                // $script:1002141907011423$
                // - 'Cause I'm inna good mood, I'mma tell you a secret! Watch the floors around here, or you're gonna fall!
                switch (selection) {
                    // $script:1002141907011424$
                    // - What's that supposed to mean?
                    case 0:
                        return 31;
                }
                return -1;
            case (31, 0):
                // $script:1002141907011425$
                // - Your foot's gonna fall! Watch the floors! Got it?
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Next,
            (30, 1) => NpcTalkButton.SelectableDistractor,
            (31, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11004302.py
""" 11004302: Ghost """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:1002141907011418$
        # - I'mma scratch <i>everything!</i>
        return None # TODO

    def __30(self, pick: int) -> int:
        if self.index == 0:
            # $script:1002141907011422$
            # - Now I'mma ghost, I can talk to people! How fun!
            return 30
        elif self.index == 1:
            # $script:1002141907011423$
            # - 'Cause I'm inna good mood, I'mma tell you a secret! Watch the floors around here, or you're gonna fall!
            if pick == 0:
                # $script:1002141907011424$
                # - What's that supposed to mean?
                return 31
            return -1
        return -1

    def __31(self, pick: int) -> int:
        # $script:1002141907011425$
        # - Your foot's gonna fall! Watch the floors! Got it?
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.NEXT
        elif (self.state, self.index) == (30, 1):
            return Option.SELECTABLE_DISTRACTOR
        elif (self.state, self.index) == (31, 0):
            return Option.CLOSE
        return Option.NONE
